src/spec-node/featuresCLI/publishCommandImpl.ts

```diff
@@ -0,0 +1,92 @@
+import { fetchRegistryAuthToken, HEADERS } from '../../spec-configuration/containerFeaturesOCI';
+import { request } from '../../spec-utils/httpRequest';
+import { Log, LogLevel } from '../../spec-utils/log';
+
+const semverCompare = require('semver-compare');
+const semver = require('semver');
+
+interface versions {
+    name: string;
+    tags: string[];
+}
+
+export async function getPublishedVersions(featureId: string, registry: string, namespace: string, output: Log) {
+    try {
+        const url = `https://${registry}/v2/${namespace}/${featureId}/tags/list`;
```

Comment:
You'll be able to remove everything in this block, in favor of calling

[`export async function pushOCIFeature(output: Log, featureRef: OCIFeatureRef, pathToTgz: string, tags: string[]): Promise<boolean>`](https://github.com/devcontainers/cli/pull/99/files#diff-290d9da09161cfae34c9229fe5d71d96446a8282fd39a384cf82a72b40418940R12)

and

[`export async function pushFeatureCollectionMetadata(output: Log, featureCollectionRef: OCIFeatureCollectionRef, pathToCollectionJson: string): Promise<boolean> {
`](https://github.com/devcontainers/cli/pull/99/files#diff-290d9da09161cfae34c9229fe5d71d96446a8282fd39a384cf82a72b40418940R12)

Reply:
Perhaps we can save this for a future PR, since #99 makes significant changes to the `containerFeaturesOCI.ts` file

Reply:
Moved!